pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+
+        return_value = float(display_content[3:11])
+        if display_content[11:13] == "GZ":
+            return_value *= 1e9
+        else:
+            return_value *= 1e6
+
+        return return_value
+
+    @frequency.setter
+    def frequency(self, frequency):
+        self.write("FR%08.4fGZ" % (float(frequency) / 1e9))
+        self.check_errors()
+
+    limits_enabled = Instrument.control(
+        "SM", "LM%d",
+        """
+        Control the limits checking function to allow the power meter to monitor the
+        power level at the sensor and to indicate when that power is outside
+        preset limits.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.LimitsCheckingStatus),
+        check_set_errors=True
+    )
+
+    @property
+    def limit_high(self):
+        """
+        Control the upper limit for the builtin limit checking.
+        """
+        self.write("LH")
+        # returns HI +299.999dB
+        display_content = self.display_output
+        assert display_content[0:2] == "HI"
+        self.write("EX")
+        return float(display_content[3:11])
+
+    @limit_high.setter
+    def limit_high(self, limit):
+        """
+        Control the upper limit for the builtin limit checking.
+        """
+        values = [-299.999, 299.999]
+        strict_range(limit, values)
+
+        self.write("LH%7.3fEN" % limit)
+        self.check_errors()
+
+    @property
+    def limit_low(self):
+        """
+        Control the lower limit for the builtin limit checking.
+        """
+        self.write("LL")
+        # returns HI +299.999dB
+        display_content = self.display_output
+        assert display_content[0:2] == "LO"
+        self.write("EX")
+        return float(display_content[3:11])
+
+    @limit_low.setter
+    def limit_low(self, limit):
+        """
+        Control the lower limit for the builtin limit checking.
+        """
+        values = [-299.999, 299.999]
+        strict_range(limit, values)
+
+        self.write("LL%7.3fEN" % limit)
+        self.check_errors()
+
+    limit_high_hit = Instrument.measurement(
+        "SM",
+        """
+        Get if the upper limit check got triggered.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.LimitsStatus),
+    )
+
+    limit_low_hit = Instrument.measurement(
+        "SM",
+        """
+        Get if the lower limit check got triggered.
+        """,
+        map_values=True,
+        values={True: 2, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.LimitsStatus),
+    )
+
+    power = Instrument.measurement(
+        "",
+        """
+        Measure the power at the power sensor attached to the power meter in the corresponding unit.
+        In case a measurement would be invalid the power meter responds with the value +9.0200e+40.
```

Comment:
You could add a get_process, which replaces that number with `float("nan")`, if you like.